supervision/metrics/detection.py

```diff
@@ -216,20 +216,28 @@ def _validate_input_tensors(
                 f"Number of predictions ({len(predictions)}) and targets ({len(targets)}) must be equal."
             )
         if len(predictions) > 0:
-            if not isinstance(predictions[0], np.ndarray) or not isinstance(
-                targets[0], np.ndarray
-            ):
-                raise ValueError(
-                    f"Predictions and targets must be lists of numpy arrays. Got {type(predictions[0])} and {type(targets[0])} instead."
-                )
-            if predictions[0].shape[1] != 6:
-                raise ValueError(
-                    f"Predictions must have shape (N, 6). Got {predictions[0].shape} instead."
-                )
-            if targets[0].shape[1] != 5:
-                raise ValueError(
-                    f"Targets must have shape (N, 5). Got {targets[0].shape} instead."
-                )
+            first_non_empty_prediction = next(
```

Comment:
@kirilllzaitsev, why do we need to look for the first non-empty?

Reply:
Nope,  I found the real issue.

Reply:
@SkalskiP, prediction/target can be either empty or non-empty. Checking the first non-empty instance and assuming that all subsequent non-empty instances will have the same shape, I can raise ValueError if the shape is wrong.